src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -513,7 +384,11 @@ async def resolve_parsed_dependencies(
         owners_per_import = await MultiGet(
             Get(
                 PythonModuleOwners,
-                PythonModuleOwnersRequest(imported_module, resolve=request.resolve),
+                PythonModuleOwnersRequest(
+                    imported_module,
+                    resolve=request.resolve,
+                    locality=request.field_set.address.spec_path,
```

Comment:
As mentioned on the ticket, it would be good if the requesting path was only set if this setting is enabled: otherwise, the satisfying `@rule` will re-run for each requesting subdirectory.

That might also be an argument for actually using the sourceroot here instead: if you have thousands of unique requesting directories, you might still only have hundreds to tens of unique sourceroots.

Reply:
True. Although no processes will be invalidated, so this is just a one-more-rule-deep memoization hit. But still probably measurable. 

That said, we set the requesting path in 4 different places, and consolidating that logic into a `@rule_helper` will incur a similar rule graph cost. But I guess that `@rule_helper` will be a lot faster than rerunning the underlying `@rule`.